src/query/sql/src/planner/plans/scan.rs

```diff
@@ -165,6 +174,57 @@ impl Scan {
         used_columns.extend(self.columns.iter());
         used_columns
     }
+
+    pub fn sample_filter(&self, stats: &Option<TableStatistics>) -> Result<Option<ScalarExpr>> {
```

Comment:
SampleLevel should help us eliminate block scan.
 
So we can add `hash(_block_name) % 100`  as `probability` for fuse engine.